components/threading/virtualized_thread_viewer/virtualized_thread_viewer.tsx

```diff
@@ -0,0 +1,379 @@
+        } else {
+            postIndex = getNewMessageIndex(this.props.replyListIds);
+        }
+
+        return postIndex === -1 ? 0 : postIndex;
+    }
+
+    scrollToItem = (index: number, position: string, offset?: number) => {
+        if (this.listRef.current) {
+            this.listRef.current.scrollToItem(index, position, offset);
+        }
+    }
+
+    scrollToBottom = () => {
+        if (this.canScrollToBottom()) {
+            this.scrollToItem(0, 'end');
+        }
+    }
+
+    scrollToNewMessage = () => {
+        this.scrollToItem(getNewMessageIndex(this.props.replyListIds), 'start', OFFSET_TO_SHOW_TOAST);
+    }
+
+    scrollToHighlightedPost = () => {
+        const {highlightedPostId, replyListIds} = this.props;
+
+        if (highlightedPostId) {
+            this.scrollToItem(replyListIds.indexOf(highlightedPostId), 'center');
+        }
+    }
+
+    handleScrollStop = () => {
+        if (this.mounted) {
+            this.setState({isScrolling: false});
+        }
+    }
+
+    renderRow = memoize(({data, itemId, style}: {data: any; itemId: any; style: any}) => {
+        const index = data.indexOf(itemId);
+        let className = '';
+        let a11Index = 0;
+        const basePaddingClass = 'post-row__padding';
+        const previousItemId = (index !== -1 && index < data.length - 1) ? data[index + 1] : '';
+        const nextItemId = (index > 0 && index < data.length) ? data[index - 1] : '';
+
+        if (isDateLine(nextItemId) || isStartOfNewMessages(nextItemId)) {
+            className += basePaddingClass + ' bottom';
+        }
+
+        if (isDateLine(previousItemId) || isStartOfNewMessages(previousItemId)) {
+            if (className.includes(basePaddingClass)) {
+                className += ' top';
+            } else {
+                className += basePaddingClass + ' top';
+            }
+        }
+
+        if (!isDateLine(itemId) && !isStartOfNewMessages(itemId)) {
+            a11Index++;
+        }
+
+        const isLastPost = itemId === this.latestPostId;
+        const isFirstPost = itemId === this.props.selected.id;
+
+        return (
+            <div
+                style={style}
+                className={className}
+            >
+                <Row
+                    a11Index={a11Index}
+                    currentUserId={this.props.currentUserId}
+                    isFirstPost={isFirstPost}
+                    isLastPost={isLastPost}
+                    listId={itemId}
+                    onCardClick={this.props.onCardClick}
+                    onCardClickPost={this.props.onCardClickPost}
+                    previousPostId={getPreviousPostId(data, index)}
+                    teamId={this.props.teamId}
+                    timestampProps={this.props.useRelativeTimestamp ? THREADING_TIME : undefined}
+                />
+                {isLastPost && (
+                    <CreateComment
```

Comment:
> Does the CreateComment need to be part of the list? It seems a bit weird to have it in the infinite list.

Hm, I'll add a `create-comment` id in the list so we can add it like the other components.

> This might be outside of the scope of this PR, but at some point, we actually wanted it to be sticky and always visible on the screen. We ran into some issues with the way menus and such overlaps (and possibly some other issues that I'm forgetting), but making CreateComment sticky was preferred.

It always felt weird to have the create comment inside the scrollable list. If that was the case I'm more than willing to add a task so we can revisit the position, @matthewbirtch, @esethna what do you say?

Reply:
As long as its in the list, I'd be okay to keep doing it with `isLastPost` instead of adding an ID to the list. They both seem like good ways to do it.

Reply:
oh, I just realized that having it render like that (with `isLastPost`) un-mounts and re-mounts `CreateComment` when a new message arrives, which messes focusing the textbox.

Reply:
> It always felt weird to have the create comment inside the scrollable list. If that was the case I'm more than willing to add a task so we can revisit the position, @matthewbirtch, @esethna what do you say?

@koox00 yes! A follow-up ticket to look at this would be awesome. This has bothered me for a long time

Reply:
@koox00 Were you able to figure out how to fix the CreateComment remounting?

Reply:
@hmhealey I've just added a commit that makes `create-comment` a part of the list ids, much like `start-of-new-messages` and `dateline-` is.